test/jdk/java/util/HashMap/WhiteBoxResizeTest.java

```diff
@@ -336,7 +339,25 @@ List<Object[]> genRequestedSizeCases(int size, int cap) {
         return Arrays.asList(
                 rsc("rshm", size, cap, () -> HashMap.newHashMap(size)),
                 rsc("rslm", size, cap, () -> LinkedHashMap.newLinkedHashMap(size)),
-                rsc("rswm", size, cap, () -> WeakHashMap.newWeakHashMap(size))
+                rsc("rswm", size, cap, () -> WeakHashMap.newWeakHashMap(size)),
+                rsc("rshs", size, cap, () -> {
+                    try {
+                        Field mapField = HashSet.class.getDeclaredField("map");
+                        HashSet<String> set = HashSet.newHashSet(size);
+                        return (Map<String, String>) mapField.get(set);
+                    } catch (Exception e) {
+                        throw new RuntimeException(e);
+                    }
+                }),
+                rsc("rsls", size, cap, () -> {
+                    try {
+                        Field mapField = HashSet.class.getDeclaredField("map");
+                        LinkedHashSet<String> set = LinkedHashSet.newLinkedHashSet(size);
+                        return (Map<String, String>) mapField.get(set);
+                    } catch (Exception e) {
+                        throw new RuntimeException(e);
+                    }
+                })
```

Comment:
These probably need a `mapField.setAccessible(true)` call, or a `VarHandle` for the `HashSet.map` field.

Reply:
Yes, this test fails with IllegalAccessException. Probably it's easiest to use a VarHandle to get private fields, similar to other usage already in this test.

This test case is a bit odd though in that it's supposed to test HashSet and LinkedHashSet but it mostly actually tests HashMap. It creates the Set instance and immediately extracts the HashMap, which is then passed to the actual test, which operates directly on the HashMap. It would be preferable to create a Set; add an element (so that it's properly allocated); and then make assertions over the Set (which involve extracting the HashMap, etc.) It seems like there should be factoring that allows this sort of arrangement to be retrofitted without adding too much complication.

Finally, please add "8284780" to the `@bug` line at the top of this test.

Reply:
> Yes, this test fails with IllegalAccessException. Probably it's easiest to use a VarHandle to get private fields, similar to other usage already in this test.
> 
> This test case is a bit odd though in that it's supposed to test HashSet and LinkedHashSet but it mostly actually tests HashMap. It creates the Set instance and immediately extracts the HashMap, which is then passed to the actual test, which operates directly on the HashMap.

Considering about it being a whitebox test, it is not very weird, but still weird enough.
A better way might be wrap it into another layer.
Will try it when I have time.

Reply:
@stuart-marks I refactored the tests. please have a look.

Reply:
@stuart-marks 8284780 added.